migration_original/ODS1Stage/tables/Base/ProviderToDisplaySpecialty/spu_translated_ProviderToDisplaySpecialty.sql

```diff
@@ -0,0 +1,169 @@
+    truncate_statement STRING;
+    select_statement STRING; -- CTE and Select statement for the insert
+    insert_statement STRING; -- Insert statement 
+    status STRING; -- Status monitoring
+   
+---------------------------------------------------------
+--------------- 2.Conditionals if any -------------------
+---------------------------------------------------------   
+   
+BEGIN
+    IF (IsProviderDeltaProcessing) THEN
+           delete_statement := 'DELETE FROM Base.ProviderToDisplaySpecialty 
+                                    WHERE ProviderID IN 
+                                        (SELECT ProviderID 
+                                         FROM Raw.ProviderDeltaProcessing)';
+
+           select_statement := '
+           WITH CTE_ProviderBatch AS (
+                SELECT ProviderId
+                FROM Raw.ProviderDeltaProcessing
+            ),';
+    ELSE
+           truncate_statement := 'TRUNCATE TABLE Base.ProviderToDisplaySpecialty';
+           
+           select_statement := '
+           WITH CTE_ProviderBatch AS (
+                SELECT ProviderId
+                FROM Base.Provider),';
+    END IF;
+
+
+---------------------------------------------------------
+----------------- 3. SQL Statements ---------------------
+---------------------------------------------------------     
+
+--- Select Statement
+
+-- If conditionals:
+select_statement := select_statement || 
+                    $$ CTE_ProviderBatch AS (
+                                SELECT ProviderId
+                                FROM Base.Provider),
+                                
+                            NotExists AS (
+                              SELECT 1
+                              FROM Base.ProviderToSpecialty pts
+                              LEFT JOIN Base.DisplaySpecialtyRule AS dsr ON pts.specialtyid = dsr.specialtyid
+                              LEFT JOIN Base.Provider AS P ON p.providerid = pts.providerid
+                              LEFT JOIN Base.DisplaySpecialtyRuleToSpecialty dsrs
+                                ON dsrs.DisplaySpecialtyRuleID = dsr.DisplaySpecialtyRuleID
+                                AND dsrs.SpecialtyID = pts.SpecialtyID
+                              WHERE pts.ProviderID = p.ProviderID
+                                AND pts.IsSearchableCalculated = 1
+                                AND dsrs.SpecialtyID IS NULL
+                            ),
+                            CTE_ProviderDisplay AS (
+                              SELECT DISTINCT c.DisplaySpecialtyRuleID, a.ProviderID
+                              FROM Base.ProviderToSpecialty a
+                              JOIN CTE_ProviderBatch b ON b.ProviderID = a.ProviderID
+                              JOIN Base.DisplaySpecialtyRule c ON c.SpecialtyID = a.SpecialtyID
+                              WHERE a.IsSearchableCalculated = 1 AND
+                                    NOT EXISTS (SELECT * FROM NotExists)
```

Comment:
add cte_ to NotExists
```suggestion
                                    NOT EXISTS (SELECT * FROM CTE_NotExists)
```